middlewares/retry.go

```diff
@@ -32,12 +33,19 @@ func NewRetry(attempts int, next http.Handler) *Retry {
 }
 
 func (retry *Retry) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	// if we might make multiple attempts, swap the body for an ioutil.NopCloser
+	// cf https://github.com/containous/traefik/issues/1008
+	if retry.attempts > 1 {
+		body := r.Body
+		defer body.Close()
+		r.Body = ioutil.NopCloser(body)
+	}
 	attempts := 1
 	for {
 		recorder := NewRecorder()
 		recorder.responseWriter = rw
 		retry.next.ServeHTTP(recorder, r)
-		if !isNetworkError(recorder.Code) || attempts >= retry.attempts {
+		if recorder.Error == nil || attempts >= retry.attempts {
```

Comment:
@bamarni are you sure you only have network errors in there? Maybe I'm wrong but I don't see the filter anymore.